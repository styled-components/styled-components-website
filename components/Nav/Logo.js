import styled, { css } from 'styled-components';

import rem from '../../utils/rem';

const Logo = styled.div`
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;

  background-position: center;
  background-size: contain;

  ${p =>
    p.compact
      ? css`
          background-image: url(/icon.png);
          width: ${rem(100)};
          height: ${rem(30)};
        `
      : css`
          background-image: url(/nav-logo.png);
          width: ${rem(164)};
          height: ${rem(40)};
        `};
`;

export default Logo;
