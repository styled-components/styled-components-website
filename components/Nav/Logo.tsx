import styled from 'styled-components';
import rem from '../../utils/rem';

const Logo = styled.div`
  background-image: url(/nav-logo.png);
  background-position: center;
  background-size: contain;
  box-sizing: border-box;
  display: inline-block;
  height: ${rem(40)};
  width: ${rem(164)};
  vertical-align: middle;
`;

export default Logo;
