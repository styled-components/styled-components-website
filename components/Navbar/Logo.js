import styled, { css } from 'styled-components'

import rem from '../../utils/rem'
import { mobile } from '../../utils/media'

const Logo = styled.div`
  display: inline-block;
  width: 100%;
  height: 100%;

  box-sizing: border-box;

  background-image: url(/static/logo.png);
  background-size: contain;
  background-position: center;

  ${mobile(css`
    background-image: url(/static/icon.png);
    background-position: center;
    width: ${rem(70)};
    height: ${rem(70)};
  `)}
`

export default Logo
