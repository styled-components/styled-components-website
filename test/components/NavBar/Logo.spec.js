import React from 'react';
import renderer from 'react-test-renderer';
import Logo from '../../../components/Nav/Logo';

test('Logo renders correctly', () => {
  const tree = renderer.create(<Logo />).toJSON();

  expect(tree).toMatchSnapshot();
});
