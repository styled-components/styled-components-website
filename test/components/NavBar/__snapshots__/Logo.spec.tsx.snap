// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
  background-position: center;
  background-size: contain;
  background-image: url(/nav-logo.png);
  width: 9.11111111111111rem;
  height: 2.2222222222222223rem;
}

<div
  className="c0"
/>
`;
